### PR TITLE
Build api

### PR DESCRIPTION
**Pull request by Vbehal01**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+from fastapi import FastAPI, UploadFile, HTTPException
+import os
+from fastapi.responses import JSONResponse
+from deepface import DeepFace
+
+app = FastAPI()
+
+@app.post("/Register/")
```
Comment (ishucr7) [inline main.py]: Google "best api route naming practices"

File: main.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+def register_face(file: UploadFile):
+    try:
+        destination_folder = "dataset"
```
Comment (ishucr7) [inline main.py]: Move this to constants

File: main.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        os.makedirs(destination_folder, exist_ok=True)
+        with open(os.path.join(destination_folder, file.filename), "wb") as f:
```
Comment (ishucr7) [inline main.py:13]: No need to handle this right now.
Do it in future PR

what if someone (person A) submits abc.png and then someone else(person B) submits abc.png? 

Based on what i can see, the old image file will be overwritten. This shouldn't ideally happen.

So, how would you handle this situation?

File: main.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            f.write(file.file.read())
+
+        return {"filename": file.filename}
+    except Exception as e:
+        return HTTPException(status_code=400, detail=str(e))
+    finally:
+        pickle_file_path="/workspaces/face_recognition/dataset/representations_vgg_face.pkl"
```
Comment (ishucr7) [inline main.py:19]: Would your docker container have worskpaces folder in it?

I don't think so. So think how you'd handle this

Comment (ishucr7) [reply main.py:19]: And use constants for path.

In case you want to join two paths. Read about os.path.join

File: main.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        if os.path.exists(pickle_file_path):
+            os.remove(pickle_file_path)
```
Comment (ishucr7) [inline main.py:21]: No meed to deal with it in this PR
1) Why are you removing it?
2) what if someone makes the recognise api call as soon as you remove it? How would their recognition work?

File: main.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+@app.get("/Recognise/")
```
Comment (ishucr7) [inline main.py]: Same: route naming

File: main.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+async def recognise_face():
+    result = DeepFace.find(img_path="img4.jpg", db_path="/workspaces/face_recognition/dataset")
```
Comment (ishucr7) [inline main.py]: Why is it hardcoded and not taking input in the request?

File: main.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    ans=[]
+    for i in result:
```
Comment (ishucr7) [inline main.py]: Please use better names than a and i

File: requirements.txt [hunk 9]
```diff
@@ -0,0 +1,69 @@
+absl-py==2.0.0
+annotated-types==0.5.0
+anyio==3.7.1
+astunparse==1.6.3
+beautifulsoup4==4.12.2
```
Comment (ishucr7) [inline requirements.txt:5]: Why does it have beautifulsoup?

I think you've mixed hp different project's envs

File: .gitignore [hunk 10]
```diff
@@ -0,0 +1,3 @@
+fast_api
+__pycache__
```
Comment (ishucr7) [inline .gitignore:2]: Don't push your sample images onto github.

In case you want to, put only 1 or 2 and name them properly

File: main.py (hunk 1 above)
Comment (Vbehal01) [reply main.py]: done

File: main.py (hunk 2 above)
Comment (Vbehal01) [reply main.py]: done

File: main.py (hunk 6 above)
Comment (Vbehal01) [reply main.py]: done

File: main.py (hunk 7 above)
Comment (Vbehal01) [reply main.py]: done

File: main.py (hunk 8 above)
Comment (Vbehal01) [reply main.py]: done

File: .gitignore (hunk 10 above)
Comment (Vbehal01) [reply .gitignore:2]: done

**Pull request merged**